omod/src/main/java/org/openmrs/module/oauth2login/web/controller/CustomLogoutSuccessHandler.java

```diff
@@ -9,48 +9,32 @@
  */
 package org.openmrs.module.oauth2login.web.controller;
 
+import static org.openmrs.module.oauth2login.OAuth2LoginConstants.USER_PROP_ID_TOKEN;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.util.Properties;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
 import org.apache.commons.lang3.StringUtils;
 import org.openmrs.api.context.Context;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.security.core.Authentication;
-import org.springframework.security.oauth2.client.OAuth2RestOperations;
-import org.springframework.security.oauth2.common.OAuth2AccessToken;
 import org.springframework.security.web.authentication.logout.LogoutSuccessHandler;
 import org.springframework.security.web.authentication.logout.SimpleUrlLogoutSuccessHandler;
 
-import javax.servlet.ServletException;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import java.io.IOException;
-import java.net.URLEncoder;
-import java.util.Properties;
-
 public class CustomLogoutSuccessHandler extends SimpleUrlLogoutSuccessHandler implements LogoutSuccessHandler {
 	
-	private OAuth2RestOperations restTemplate;
-	
-	@Autowired
-	public void setRestTemplate(@Qualifier("oauth2.restTemplate") OAuth2RestOperations restTemplate) {
-		this.restTemplate = restTemplate;
-	}
-	
-	private String getToken() {
-		OAuth2AccessToken accessToken = restTemplate.getAccessToken();
-		if (accessToken != null) {
-			return StringUtils.defaultString(accessToken.getValue());
-		}
-		return StringUtils.EMPTY;
-	}
-	
 	@Override
 	public void onLogoutSuccess(HttpServletRequest request, HttpServletResponse response, Authentication authentication)
 	        throws IOException, ServletException {
 		Properties properties = OAuth2BeanFactory.getProperties(OAuth2BeanFactory.getOAuth2PropertiesPath());
 		String redirectPath = properties.getProperty("logoutUri");
 		//the redirect path can contain a [token] that should be replaced by the aut token
 		if (StringUtils.isNoneBlank(redirectPath) && redirectPath.contains("[token]")) {
-			String token = getToken();
+			String token = Context.getAuthenticatedUser().getUserProperty(USER_PROP_ID_TOKEN);
```

Comment:
By this time, do we still have an authenticated user?

Reply:
Yes, because this is called on oauth2 logout success and not OpenMRS logout success which is later called on the next few lines.